meta-avocado/recipes-avocado/packages/avocado-pkg-bootfiles.bb

```diff
@@ -0,0 +1,77 @@
+SUMMARY = "Copy boot files to /opt/avocado/bootfiles"
+LICENSE = "Apache-2.0"
+
+IMAGE_BOOTFILES ?= ""
+AVOCADO_PKG_BOOTFILES_EXTRA ?= ""
+AVOCADO_IMAGES_INSTALL_PREFIX = "${SDKPATHINSTALL}/images"
+FILES:${PN} = "${AVOCADO_IMAGES_INSTALL_PREFIX}"
+PACKAGES = "${PN}"
+PACKAGE_ARCH = "${MACHINE_ARCH}"
+
+do_compile[noexec] = "1"
+do_install[depends] += "avocado-image-rootfs:do_image_complete"
+
+fakeroot python do_install() {
+    import os
+    deploy_dir = d.getVar('DEPLOY_DIR_IMAGE')
+    install_prefix_path = d.getVar('AVOCADO_IMAGES_INSTALL_PREFIX')
+    install_dir = d.getVar('D') + install_prefix_path
+
+    # Ensure the target directory hierarchy exists
+    bb.utils.mkdirhier(install_dir)
+
+    # Attempt to set ownership of the main installation directory to root:root.
+    # This helps if intermediate directories were created with host UID/GID.
+    if install_prefix_path.strip() and install_prefix_path.strip() != "/":
+        try:
+            # Chown the final created directory.
+            # For a path like ${D}/opt/avocado/images, this chowns .../images.
+            os.chown(install_dir, 0, 0)
+        except OSError as e:
+            bb.warn(f"Could not chown install_dir {install_dir}: {e}. This might be acceptable if running under fakeroot.")
```

Comment:
I have reservations about this code block. I suspect it may lead to build failures.

Is there a prescribed way to set ownership and permissions with BitBake?